compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java

```diff
@@ -62,6 +62,7 @@ protected BIRNode(Location pos) {
         public final List<BIRFunction> functions;
         public final List<BIRAnnotation> annotations;
         public final List<BIRConstant> constants;
+        public boolean isListenerAvailable;
```

Comment:
can we make this final and pass the value from the constructor?

Reply:
I think no. because when we creating the bir package we do not know listener available. we need to visit all global variables to decide that.